arches/app/views/api/controlled_lists.py

```diff
@@ -0,0 +1,429 @@
+                status=HTTPStatus.BAD_REQUEST,
+            )
+        except ValidationError as ve:
+            return JSONErrorResponse(
+                message="\n".join(ve.messages), status=HTTPStatus.BAD_REQUEST
+            )
+
+        return JSONResponse(status=HTTPStatus.NO_CONTENT)
+
+    def delete(self, request, **kwargs):
+        item_id: UUID = kwargs.get("id")
+        objs_deleted, unused = ControlledListItem.objects.filter(pk=item_id).delete()
+        if not objs_deleted:
+            return JSONErrorResponse(status=HTTPStatus.NOT_FOUND)
+        return JSONResponse(status=HTTPStatus.NO_CONTENT)
+
+
+@method_decorator(
+    group_required("RDM Administrator", raise_exception=True), name="dispatch"
+)
+class ControlledListItemValueView(View):
+    def add_new_value(self, request):
+        data = JSONDeserializer().deserialize(request.body)
+
+        value = ControlledListItemValue(
+            controlled_list_item_id=UUID(data["item_id"]),
+            valuetype_id=data["valuetype_id"],
+            language_id=data["language_id"],
+            value=data["value"],
+        )
+        try:
+            value.full_clean()
+        except ValidationError as ve:
+            return JSONErrorResponse(
+                message="\n".join(ve.messages), status=HTTPStatus.BAD_REQUEST
+            )
+        value.save()
+
+        return JSONResponse(value.serialize(), status=HTTPStatus.CREATED)
+
+    def post(self, request, **kwargs):
+        if not (value_id := kwargs.get("id", None)):
+            return self.add_new_value(request)
+
+        data = JSONDeserializer().deserialize(request.body)
+
+        try:
+            value = ControlledListItemValue.objects.values_without_images().get(
+                pk=value_id
+            )
+        except ControlledListItemValue.DoesNotExist:
+            return JSONErrorResponse(status=HTTPStatus.NOT_FOUND)
+
+        value.value = data["value"]
+        value.valuetype_id = data["valuetype_id"]
+        try:
+            value.language = Language.objects.get(code=data["language_id"])
+        except Language.DoesNotExist:
+            return JSONErrorResponse(status=HTTPStatus.NOT_FOUND)
+
+        try:
+            value.full_clean()
+        except ValidationError as ve:
+            return JSONErrorResponse(
+                message="\n".join(ve.messages), status=HTTPStatus.BAD_REQUEST
+            )
+        value.save()
+
+        return JSONResponse(value.serialize())
+
+    def delete(self, request, **kwargs):
+        value_id = kwargs.get("id")
+        try:
+            value = ControlledListItemValue.objects.values_without_images().get(
+                pk=value_id
+            )
+        except:
+            return JSONErrorResponse(status=HTTPStatus.NOT_FOUND)
+
+        try:
+            value.delete()
+        except ValidationError as ve:
+            return JSONErrorResponse(
+                message="\n".join(ve.messages), status=HTTPStatus.BAD_REQUEST
```

Comment:
I know this is supper nit-picky, but wouldn't the request status be best described as:
```suggestion
                message="\n".join(ve.messages), status=HTTPStatus.FORBIDDEN
```

Reply:
403 Forbidden implies an authenticated user with insufficient perms, but here we have an application logic issue -- this catches the error raised by ControlledListItemValue.delete() if a user tries to delte the item's only remaining pref label. So I think 400 is right.

Reply:
[Looking at it in more detail](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403), you're definitely right.